src/ORMSubject.php

```diff
@@ -50,7 +51,11 @@ public function notify($entitySource, $event, $data, $oldData = null) {
         }
         foreach ((array)$this->observers[$entitySource] as $observer) {
             $observer->log("Observer: notifying " . $observer->getMapper()->getTable() . ", changes in $entitySource");
-            $observer->getObserverdProcessor()->process(new ObserverData($entitySource, $event, $data, $oldData, $observer->getRepository()));
+            try {
+                $observer->getObserverdProcessor()->process(new ObserverData($entitySource, $event, $data, $oldData, $observer->getRepository()));
+            } catch (Throwable $e) {
+                $observer->getObserverdProcessor()->onError(new ObserverOnErrorData($e, $data, $oldData));
```

Comment:
In some situations, it is important to know what event threw the error. 

Suggestion:

1. Create the $observerData before call the process.
2. If an error happened just call the $observerData the onError metodo alongside the exception. 

```suggestion
                $observer->getObserverdProcessor()->onError($e, $observerData);
```